test/runtests.jl

```diff
@@ -37,6 +37,7 @@ tests = [
 
 const testdir = dirname(@__FILE__)
 
+push!(ChainRulesTestUtils.TRANSFORMS_TO_ALT_TANGENTS, x -> @thunk(x))
```

Comment:
This seems like an undocumented API…

If this feature is going to be required in ChainRules.jl, shouldn't ChainRulesTestUtils.jl implement a documented API to test it?

Reply:
A comment  should be added here to explain what is going on
(and perhaps more clear docs, it is mentioned [here](https://juliadiff.org/ChainRulesTestUtils.jl/dev/api.html#ChainRulesTestUtils.test_rrule-Tuple) but not about mutating it).

This is the soft-way of rolling out the changes.
ChainRulesTestUtils v1.0 will define 
`ChainRulesTestUtils.TRANSFORMS_TO_ALT_TANGENTS = [x -> @thunk(x), x->ZeroTangent(), ...]`.
But right now it is set to `[]`.
And then as we go and fix packages we can add this line to make sure it gets those new tests.
So that we can roll that out one package at a time, without breaking every other package.

In a normal package this would be unsafe, since each package mutating it could be seen by each other.
But ChainRulesTestUtils is only ever a direct, test-time, dependency.
So I believe it is safe.

The 1.0 API will mean this doesn't need to be touched and this line will have no effect.

Reply:
I came here to say what Lyndon said. Will add a comment explaining this temporary state

Reply:
I agree that this test should be opt-in, but it seems like it would be cleaner if you could just opt-in by something like:
```jl
ChainRulesTestUtils.enable_thunk_testing()
```
and this would become a no-op in 1.0.

Reply:
That's true.
I will make that change and then we can update this